nbconverted/3_park_analysis.py

```diff
@@ -0,0 +1,377 @@
+# coding: utf-8
+
+# ## Re-analysis of Park et al. findings using gene expression-based mutation signatures
+# 
+# In [Park et al. 2021](https://www.nature.com/articles/s41467-021-27242-3), the authors study interactions between point mutations and CNVs in TCGA data. Essentially, they do an overrepresentation analysis to identify genes/cancer types that have more samples with _both_ a point mutation and a CNV than expected by chance, implying that these genes tend to require "two hits" (the point mutation and the CNV in this case) to be activated (oncogenes) or inactivated (tumor suppressors).
+# 
+# Here, we want to take the genes/cancer types they identified, and analyze the functional effects in the context of our mutation prediction classifiers. Our hypothesis is that in the "two-hit" genes, samples with "two hits" (a point mutation and a CNV) will have a higher predicted mutation probability than samples with zero or one hit.
+
+# In[1]:
+
+
+from pathlib import Path
+import pickle as pkl
+
+import numpy as np
+import pandas as pd
+import seaborn as sns
+import matplotlib.pyplot as plt
+
+import sys; sys.path.append('..')
+import config as cfg
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+
+# In[2]:
+
+
+# park geneset info
+park_loss_data = cfg.data_dir / 'park_loss_df.tsv'
+park_gain_data = cfg.data_dir / 'park_gain_df.tsv'
+
+# park gene/cancer type predictions
+park_preds_dir = cfg.data_dir / 'park_genes_preds'
+
+# mutation and copy number data
+pancancer_pickle = Path('/home/jake/research/mpmp/data/pancancer_data.pkl')
+
+
+# ### Load mutation info
+# 
+# For now, just use binary mutation status from the pancancer repo. In the future we could pull more granular info from MC3, but it would take some engineering of `1_get_mutation_counts` to do this for lots of genes.
+
+# In[3]:
+
+
+park_loss_df = pd.read_csv(park_loss_data, sep='\t', index_col=0)
+park_loss_df.head()
+
+
+# In[4]:
+
+
+park_gain_df = pd.read_csv(park_gain_data, sep='\t', index_col=0)
+park_gain_df.head()
+
+
+# In[5]:
+
+
+with open(pancancer_pickle, 'rb') as f:
+    pancancer_data = pkl.load(f)
+
+
+# In[6]:
+
+
+mutation_df = pancancer_data[1]
+print(mutation_df.shape)
+mutation_df.iloc[:5, :5]
+
+
+# ### Load copy number info
+# 
+# Get copy loss/gain info directly from GISTIC "thresholded" output. This should be the same as (or very similar to) what the Park et al. study uses.
+
+# In[7]:
+
+
+# we use the data source and preprocessing code from the pancancer repo, here:
+# https://github.com/greenelab/pancancer/blob/d1b3de7fa387d0a44d0a4468b0ac30918ed66886/scripts/initialize/process_copynumber.py#L21
+
+copy_thresh_df = (
+    pd.read_csv(Path('data', 'pancan_GISTIC_threshold.tsv'),
+                sep='\t', index_col=0)
+      .drop(columns=['Locus ID', 'Cytoband'])
+)
+copy_thresh_df.columns = copy_thresh_df.columns.str[0:15]
+
+print(copy_thresh_df.shape)
+copy_thresh_df.iloc[:5, :5]
```

Comment:
Is it weird that sample TCGA-OR-A5J4-01 has CNVs at all five of the alphabetically first genes? Or am I reading this table totally wrong?

Reply:
No, you're reading it correctly. It is a bit weird, but I guess it's plausible that some tumors just have widespread chromosomal instability [e.g. chromoplexy](https://en.wikipedia.org/wiki/Chromoplexy).

We should probably do some basic QC of the CNV data, though - created #4 to do this in the future.